File: code/python/Card.py
class Card:
    def getSuit(self)->str:
        return self.suit
    def getValue(self)->str:
        return self.value
    def getName(self)->str:
        return self.getValue()+"_"+self.getSuit()
    def setSuit(self, s:str):
        if len(s)==1:
            suits={
                'c':"Clubs", 
                'h':"Hearts", 
                's':"Spades",
                'd':"Diamonds",
                'C':"Clubs",
                "H":"Hearts",
                "S":"Spades",
                "D":"Diamonds"
            }
            self.suit=suits.get(s, "InvalidCardSuit")
        else:
            #add error checking (suit should be in form Spades, Hearts...)
            self.suit=s
    
    def setValue(self, val):
        if isinstance(val, int):
            vals={
                1:"Ace", 
                2:"Two", 
                3:"Three", 
                4:"Four", 
                5:"Five", 
                6:"Six", 
                7:"Seven", 
                8:"Eight", 
                9:"Nine", 
                10:"Ten", 
                11:"Jack", 
                12:"Queen", 
                13:"King",
                14:"Ace"
            }
            self.value=vals.get(val, "InvalidCardValue")
        elif len(val)==1:
            #convert to full string
            pass
        elif isinstance(val, str):
            #add error checking
            self.value=val
        
    def __init__(self, val='', suit=''):
        self.setValue(val)
        self.setSuit(suit)

File: code/hw4/Card.cpp
#include "Card.h"

void Card::setSuit(char s){
    switch(s){
        case 'H':
            suit="Hearts";
            break;
        case 'h':
            suit="Hearts";
            break;
        case 'S':
            suit="Spades";
            break;
        case 's':
            suit="Spades";
            break;
        case 'C':
            suit="Clubs";
            break;
        case 'c':
            suit="Clubs";
            break;
        case 'D':
            suit="Diamonds";
            break;
        case 'd':
            suit="Diamonds";
            break;
        default:
            suit="InvalidCardSuit";
            break;
    }
}
void Card::setValue(int v){
    switch (v)
    {
    case 1:
        value="Ace";
        break;
    case 2:
        value="Two";
        break;
    case 3:
        value="Three";
        break;
    case 4:
        value="Four";
        break;
    case 5:
        value="Five";
        break;
    case 6:
        value="Six";
        break;
    case 7:
        value="Seven";
        break;
    case 8:
        value="Eight";
        break;
    case 9:
        value="Nine";
        break;
    case 10:
        value="Ten";
        break;
    case 11:
        value="Jack";
        break;
    case 12:
        value="Queen";
        break;
    case 13:
        value="King";
        break;
    
    default:
        value="InvalidCardValue";
        break;
    }
}

void Card::setSuit(std::string s){
    suit=s;
}
std::string Card::getName(){
    return value+"_"+suit;
}
std::string Card::getSuit(){return suit;}
std::string Card::getRank(){return value;}
Card::Card(){}
Card::Card(int v, char s){
    Card::setSuit(s);
    Card::setValue(v);
}
Card::~Card(){
    //do onthing?
}

